app/views/partials/foyer/ressources/year-moins-2.html

```diff
@@ -4,42 +4,57 @@
     Ces informations se trouvent sur votre déclaration de revenus d'impôts {{ yearMoins2 }}.
     <br>Vous pouvez la retrouver en ligne sur <a href="http://www.impots.gouv.fr/">impots.gouv.fr</a>.
   </p>
-  <form class="form-horizontal" novalidate ng-submit="submit()">
+  <form class="form-horizontal" novalidate name="ym2form" ng-submit="submit(ym2form)">
     <div class="row">
       <div class="col-md-12" ng-repeat="individuRef in individuRefsToDisplay">
         <h2>{{ individuRef.label }}</h2>
-        <div class="form-group" ng-repeat="rnc in individuRef.rnc">
-          <label class="control-label col-sm-3" for="{{ rnc.categorie.id }}-{{ individuRef.individu.role }}">
+        <div
+          class="form-group"
+          ng-repeat="rnc in individuRef.rnc"
+          ng-init="input_name = rnc.categorie.id +  '-' + individuRef.individu.role"
+          ng-class="{ 'has-error': ym2form.$submitted && ym2form[input_name].$invalid }"
+          >
+          <label class="control-label col-sm-3" for="{{ input_name }}">
             {{ rnc.categorie.label }}
           </label>
           <div class="col-sm-3">
             <span class="input-group">
               <input
                 type="number"
-                id="{{ rnc.categorie.id }}-{{ individuRef.individu.role }}"
+                id="{{ input_name }}"
+                name="{{ input_name }}"
                 class="form-control text-right"
                 ng-model="rnc.montant"
                 min="0"
                 select-on-focus
-                zero-to-empty>
+                zero-to-empty
+                aria-describedby="help-{{ input_name}}">
               <span class="input-group-addon hidden-xs">€</span>
             </span>
+            <ul class="help-block ym2-error">
```

Comment:
I wanted to use CSS to show and hide DOM elements but that was not a good idea (done differently everywhere else). I reverted that change in a `fixup` commit and used the _standard_ method (ie. `ng-if`/`ng-show`).